collector/lib/CollectorConfig.cpp

```diff
@@ -400,6 +406,48 @@ void CollectorConfig::HandleSinspEnvVars() {
   }
 }
 
+void CollectorConfig::HandleConfigMapString(const std::string& jsonString) {
+  sensor::CollectorConfig config;
+  auto status = google::protobuf::util::JsonStringToMessage(jsonString, &config);
+
+  if (!status.ok()) {
+    CLOG(ERROR) << "Failed to parse runtime config. Error: " << status.message();
+    CLOG(ERROR) << "ConfigMap file contents: " << jsonString;
+  } else {
+    SetRuntimeConfig(config);
+    CLOG(INFO) << "Set the runtime config using a configmap";
+    CLOG(INFO) << config.DebugString();
```

Comment:
Too verbose IMO, this should be enough and easier to read:
```suggestion
    CLOG(INFO) << "Runtime configuration: " << config.DebugString();
```